geopandas/tests/test_geoseries.py

```diff
@@ -102,6 +103,14 @@ def test_align(self):
         assert a1['B'].equals(a2['B'])
         assert a1['C'].is_empty
 
+    def test_warning_if_not_aligned(self):
+        # Test that warning is issued when operating on non-aligned series
+        with pytest.warns(UserWarning):
+            self.a1.contains(self.a2)
+
+        with pytest.warns(UserWarning):
+            self.a1.union(self.a2)
```

Comment:
Can you also add a test to explicitly test that no warning is raised in the case the index is equal?

Reply:
Done :blush: